src/mlpack/core/util/param.hpp

```diff
@@ -42,27 +42,13 @@
  * should be present per binding.  BINDING_NAME should be set before calling
  * this.
  *
- * @see mlpack::IO, PARAM_FLAG(), PARAM_INT_IN(), PARAM_DOUBLE_IN(),
- * PARAM_STRING_IN(), PARAM_VECTOR_IN(), PARAM_INT_OUT(), PARAM_DOUBLE_OUT(),
- * PARAM_VECTOR_OUT(), PARAM_INT_IN_REQ(), PARAM_DOUBLE_IN_REQ(),
- * PARAM_STRING_IN_REQ(), PARAM_VECTOR_IN_REQ(), PARAM_INT_OUT_REQ(),
- * PARAM_DOUBLE_OUT_REQ(), PARAM_VECTOR_OUT_REQ(), PARAM_STRING_OUT_REQ().
- *
  * @param NAME User-friendly name.
  */
-#ifdef __COUNTER__
-  #define BINDING_USER_NAME(NAME) static \
-      mlpack::util::BindingName \
-      JOIN(io_bindingusername_dummy_object, __COUNTER__) = \
-      mlpack::util::BindingName( \
-          STRINGIFY(BINDING_NAME), NAME);
-#else
-  #define BINDING_USER_NAME(NAME) static \
-      mlpack::util::BindingName \
-      JOIN(JOIN(io_bindingusername_dummy_object, __LINE__), opt) = \
-      mlpack::util::BindingName( \
-          STRINGIFY(BINDING_NAME), NAME);
-#endif
```

Comment:
`__COUNTER__` isn't a standard macro defined by all compilers.  So I don't think that we should change this code.

Reply:
`__COUNTER__` is a gcc-specific extension that happens to also be implemented by MSVC and clang. But, those are not the only C++ compilers in the world. In the interest of keeping mlpack standards-compliant, we should not depend entirely on gcc-specific macros like that one. :+1: